cedar-lean/Cedar/Thm/Data/List.lean

```diff
@@ -1066,27 +1421,26 @@ theorem find?_pair_map {α β γ} [BEq α] (f : β → γ) (xs : List (α × β)
         (λ (x : α × β) => x.fst == k) hd tl h₁
       have h₃ := @List.find?_cons_of_neg _
         (λ (x : α × γ) => x.fst == k) (hd.fst, f hd.snd)
-        (map (fun x => (x.fst, f x.snd)) tl)
-      simp [h₁] at h₃
-      simp [h₂, h₃]
+        (map (λ x => (x.fst, f x.snd)) tl)
+      simp only [h₁, not_false_eq_true, forall_const] at h₃
+      simp only [h₂, h₃]
       exact ih
     case true =>
       have h₂ := @List.find?_cons_of_pos _
         (λ (x : α × β) => x.fst == k) hd tl h₁
       have h₃ := @List.find?_cons_of_pos _
         (λ (x : α × γ) => x.fst == k) (hd.fst, f hd.snd)
-        (map (fun x => (x.fst, f x.snd)) tl)
-      simp [h₁] at h₃
+        (map (λ x => (x.fst, f x.snd)) tl)
+      simp only [h₁, forall_const] at h₃
       simp [h₂, h₃]
 
 theorem find?_fst_map_implies_find? {α β γ} [BEq α] {f : β → γ} {xs : List (α × β)} {k : α} {fx : α × γ}:
-  List.find? (fun x => x.fst == k) (List.map (Prod.map id f) xs) = .some fx  →
-  ∃ x, xs.find? (fun x => x.fst == k) = .some x ∧ fx = Prod.map id f x
+  List.find? (λ x => x.fst == k) (List.map (Prod.map id f) xs) = .some fx  →
+  ∃ x, xs.find? (λ x => x.fst == k) = .some x ∧ fx = Prod.map id f x
 := by
   intro h
   induction xs
-  case nil =>
-    simp only [map_nil, find?_nil] at h
+  case nil => simp at h
```

Comment:
I'd prefer to leave `simp only` in place, with the goal of eventually replacing (nearly) all `simp`s with `simp only` (as we encounter them).  They are uglier, admittedly, but really good for proof stability.  I'm guessing they improve efficiency too, since search is limited to the `only` lemmas.

Reply:
Fixed, and see comment thread above